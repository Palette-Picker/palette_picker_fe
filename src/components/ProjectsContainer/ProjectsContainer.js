import React from 'react';
import ProjectCard from '../ProjectCard/ProjectCard';
import './ProjectsContainer.scss';

const ProjectsContainer = ({ projects, handleModal, handleKeyPress }) => {
  const projCards = projects.map(project => {
    return <ProjectCard
      key={project.id}
      project={project}
      handleKeyPress={handleKeyPress}
      handleModal={handleModal} />
  })
  return (
    <section className='projects'>
      { projCards }
    </section>
  )

};

export default ProjectsContainer;