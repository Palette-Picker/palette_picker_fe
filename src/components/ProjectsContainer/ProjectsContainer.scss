@import './src/variables.scss';

.projects {
  height: 100%;
  background-color: $extra-light-color;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
}

h5 {
  width: 100%;
  font-size: 1.2em;
  margin-bottom: -10px;
}