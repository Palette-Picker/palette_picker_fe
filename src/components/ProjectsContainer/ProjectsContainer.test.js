import React from 'react';
import { shallow } from 'enzyme';
import ProjectsContainer from './ProjectsContainer';

describe('ProjectsContainer', () => {

  let wrapper;
  const mockProjects = [
    {
      id: 19,
      name: "Colors",
      palettes: [
        {
          id: 29,
          name: "Purples and greens",
          color1: "#82d173",
          color2: "#abfaa9",
          color3: "#95a3b3",
          color4: "#4c2c69",
          color5: "#42253b",
          project_id: 19,
          created_at: "2019-12-09T00:11:51.362Z",
          updated_at: "2019-12-09T00:11:51.362Z"
        }]
      }];

  const mockPass = jest.fn();
  const mockToggleModal = jest.fn();

  it('should match the snapshot', () => {
    wrapper = shallow(
      <ProjectsContainer
        projects={mockProjects}
        passPaletteNameAndColors={mockPass}
        toggleModal={mockToggleModal}
      />);
    expect(wrapper).toMatchSnapshot();
  });

});