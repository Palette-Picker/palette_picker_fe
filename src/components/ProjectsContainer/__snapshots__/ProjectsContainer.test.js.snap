// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectsContainer should match the snapshot 1`] = `
<section
  className="projects"
>
  <ProjectCard
    key="19"
    passPaletteNameAndColors={[MockFunction]}
    project={
      Object {
        "id": 19,
        "name": "Colors",
        "palettes": Array [
          Object {
            "color1": "#82d173",
            "color2": "#abfaa9",
            "color3": "#95a3b3",
            "color4": "#4c2c69",
            "color5": "#42253b",
            "created_at": "2019-12-09T00:11:51.362Z",
            "id": 29,
            "name": "Purples and greens",
            "project_id": 19,
            "updated_at": "2019-12-09T00:11:51.362Z",
          },
        ],
      }
    }
    toggleModal={[MockFunction]}
  />
</section>
`;
